### PR TITLE
Letting TDI session persist across TDI python starts

### PR DESCRIPTION
**Pull request by saynb**

### CODE REVIEW

File: tdi_python/tdicli.py [hunk 1]
```diff
@@ -2147,15 +2139,33 @@ def unload_ipython_extension(ipython):
...
 
+def destroy_tdi_python_session():
+    sts = self._driver.tdi_session_destroy(tdi_session)
+    if not sts == 0:
+        print("ERROR: destroying session")
+        self._driver.dev_config_deallocate(self.dev_config)
+        return -1
+    return 0
+
+def create_tdi_python_session():
```
Comment (satish153) [inline tdi_python/tdicli.py:2150]: What will be the device here in a multi-device system?

File: tdi_python/tdicli.py [hunk 2]
```diff
@@ -2195,6 +2206,14 @@ def start_tdi(self, in_fd, out_fd, dev_id_list, udf=None, interactive=False, cIn
 
         if hasattr(sys.modules['__main__'], "ipython_app"):
             ipython_app = getattr(sys.modules['__main__'], "ipython_app")
+
+        if hasattr(sys.modules['__main__'], "tdi_session"):
+            tdi_session = getattr(sys.modules['__main__'], "tdi_session")
+        else:
+            sts = create_tdi_python_session()
```
Comment (satish153) [inline tdi_python/tdicli.py:2213]: Currently, a session is created per device.  Are we changing that? Is there going to be only one session?

File: tdi_python/tdicli.py (hunk 1 above)
Comment (saynb) [reply tdi_python/tdicli.py]: changed now

File: tdi_python/tdicli.py (hunk 2 above)
Comment (saynb) [reply tdi_python/tdicli.py]: changed for multi-device

**Review by satish153 — APPROVED** [review]
LGTM

**Pull request merged**